pkg/cmd/util/env.go

```diff
@@ -31,3 +34,21 @@ func GetEnv(key string) (string, bool) {
 	}
 	return val, true
 }
+
+// ThreadSafeSetEnv provides locking around the call to os.Setenv
+// do an internet search like "golang setenv thread safe" and
+// read about the glibc stuff
+func ThreadSafeSetEnv(key, value string) error {
+	lock.Lock()
+	defer lock.Unlock()
+	return os.Setenv(key, value)
```

Comment:
/hold

Why do need this? We use Go.

`os.Setenv` is just an alias for `syscall.Setenv`. `syscall.Setenv` already has a mutex: https://golang.org/src/syscall/env_unix.go?s=1927:1963#L83

Reply:
Ha .... son of a gun 

Good catch @dmage ... thanks

@stevekuznetsov @bparees fyi ^^^

big false alarm